rest/processing/sentinel_hub.py

```diff
@@ -78,16 +73,18 @@ def get_request_dictionary(
                 "bounds": self.construct_input_bounds(bbox, epsg_code, geometry),
                 "data": [
                     {
-                        "type": collection.api_id,
+                        "id": node_id,
+                        "type": collection["data_collection"].api_id,
                         "dataFilter": {
                             "timeRange": {
-                                "from": from_date.isoformat(),
-                                "to": to_date.isoformat(),
+                                "from": collection["from_date"].isoformat(),
+                                "to": collection["to_date"].isoformat(),
                             },
                             "previewMode": preview_mode,
                         },
                         "processing": self.construct_data_processing(resampling_method),
                     }
+                    for node_id, collection in collections.items()
```

Comment:
This line can be really stealthy when you are in a hurry. Thus, I suggest we change this to


```python
request_data_items = []
        for node_id, collection in collections.items():
            request_data_items.append({
                "id": node_id,
                "type": collection["data_collection"].api_id,
                "dataFilter": {
                    "timeRange": {
                        "from": collection["from_date"].isoformat(),
                        "to": collection["to_date"].isoformat(),
                    },
                    "previewMode": preview_mode,
                },
                "processing": self.construct_data_processing(resampling_method),
            })

        return {
            "input": {
                "bounds": self.construct_input_bounds(bbox, epsg_code, geometry),
                "data": request_data_items,
            },
            "output": self.construct_output(width, height, mimetype),
            "evalscript": evalscript,
        }
```

Reply:
This is way better I agree, changed